src/sim_model_output.R

```diff
@@ -0,0 +1,79 @@
+## Function that generates sample submission data to pass validation checks for
+## It's not particularly efficient but it's easy to understand
+## Uses a Dirichlet distribution to generate random proportions of clades
+
+#' Simulate data in appropriate model_output format
+#' 
+#' @returns Data frame in required model-output format
+#'    of simulated data for variant nowcasting.
+#' @locations Vector of location abbreviations e.g. c("MA", ...)
+#' @alpha Parameters for rdirichlet for length(test_clades)
+#' @test_clades Clades for the weekly forecasts
+#' @target_date Forecast dates
+#' @nowcast_date Nowcast date
+#' @n_samp Number of samples per location (100 required, available for testing)
+#' @seed Set random seed value for RNG
+sim_model_output <- function(locations = c("AL", "AZ", "CA", "CO", "DE"), 
+                             alpha = c(3,3,2,2,1,1),
+                             test_clades = c("24A", "24B", "24C", "24E", "recombinant", "other"),
+                             target_date = seq(as.Date("2024-09-01"), as.Date("2024-10-12"), by = "1 days"),
+                             nowcast_date = "2024-10-02",
+                             n_samp = 100,
+                             seed = 42
+){
+  require(brms) # for data simulation
+  set.seed(seed)
+  
+  # more QA to come e.g. n_samp positive integer, etc.
+  
+  # Columns to be created for output
+  target_date_col = c()
+  output_type_id_col = c()
+  loc_col = c()
+  clade_col = c()
+  value_col = c()
+  
+  # Create observations using dirichlet distribution for proportions
+  for(loc in locations){
+    for(date in target_date){
+      for(sample_pt in 0:(n_samp-1)) {
+        # Create random proportions with prob alpha for each sample point
+        props <- rdirichlet(1, alpha = alpha)
```

Comment:
This modifies the function in two ways:

1. use `withr::local_seed()` to avoid downstream effects of internally setting the seed (it future-proofs the function call so that you can move it around without affecting other random seeds). Since withr is already part of the renv file, this should work as advertised. 
2. uses the `pkg::fn()` syntax to avoid attaching the package into the global namespace.

```suggestion
  # set the `seed` and then resets it to the state of the RNG before this ran
  # https://withr.r-lib.org/reference/with_seed.html
  withr::local_seed(seed)
  
  # more QA to come e.g. n_samp positive integer, etc.
  
  # Columns to be created for output
  target_date_col = c()
  output_type_id_col = c()
  loc_col = c()
  clade_col = c()
  value_col = c()
  
  # Create observations using dirichlet distribution for proportions
  for(loc in locations){
    for(date in target_date){
      for(sample_pt in 0:(n_samp-1)) {
        # Create random proportions with prob alpha for each sample point
        props <- brms::rdirichlet(1, alpha = alpha)
```